packages/terra-application-navigation/src/extensions/_Extension.jsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+
+import ExtensionCount from './_ExtensionCount';
+import { enableFocusStyles, disableFocusStyles, generateKeyDownSelection } from '../utils/helpers';
+
+import styles from './Extension.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * The icon to display for the associated extension action.
+   */
+  icon: PropTypes.element.isRequired,
+  /**
+   * The number of notifications to be displayed for the extension.
+   */
+  notificationCount: PropTypes.number,
+  /**
+   * Function callback for selection of the extension. Return (metaData).
+   */
+  onSelect: PropTypes.func,
+  /**
+   * The text display for the extension, either the aira label or the inline text.
+   */
+  text: PropTypes.string.isRequired,
+};
+
+const defaultProps = {
+  notificationCount: 0,
+};
+
+const Extension = ({
+  notificationCount,
+  icon,
+  text,
+  onSelect,
+}) => (
+  <div
+    role="button"
```

Comment:
Why not use a `button` element here?

Reply:
The main reason for using divs in this case was purely selection and highlight styles behaving more consistent across browsers with divs, avoiding browser specific button styling.

Reply:
As long as we're [covering all that is needed for it to behave like a button](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role), that's fine with me. Looks like we have the onClick and keydown handlers set up correctly.